lib/assets/SISScraper.rb

```diff
@@ -0,0 +1,175 @@
+require 'watir-webdriver'
+
+class SISScraper
+  ALPHA = ['A','B','C','D','E','F','G','H','I','J','K','L','M','N','O','P','Q','R','S','T','U','V','W','X','Y','Z']
+
+  #TODO: This can be moved to configs later
+  DOWNLOAD_DIR = '../../../Downloads/'
+
+  def initialize()
+    @browser = Watir::Browser.new :firefox
+  end
+
+  def download_all_sis_data(user,pword)
+    go_to_sis
+    sign_in_to_sis(user, pword)
+    navigate_to_search_from_landing_page
+    wait_while_sis_processing
+    get_all_depts.each do |dept|
+      get_all_semesters.each do |semester|
+        sem = semester.strip
+        unless (sem == 'Fall 2015' || sem == 'Summer 2015')
+          search_for_classes(dept, sem)
+          ok_more_than_20_results
+          if results?
+            if downloadable?
+              download_course_info(dept, sem)
+            else
+              Kernel.puts("#{dept}#{sem} needs manual input due to too few classes.")
+            end
+            start_a_new_search
+          else
+            Kernel.puts("#{dept}#{sem} had no classes.")
+          end
+        end
+      end
+    end
+  end
+
+  private
+
+  def go_to_sis
+    @browser = Watir::Browser.new
+    @browser.goto "sis.case.edu"
+    Kernel.puts "Went to SIS"
+  end
+
+  def sign_in_to_sis(user, pword)
+    # Sign In
+    @browser.text_fields.first.wait_until_present
+    @browser.text_fields.first.set user
+    @browser.text_fields.last.wait_until_present
+    @browser.text_fields.last.set pword
+    @browser.button(:text => 'Sign In').click
+    Kernel.puts "Logged into SIS"
+  end
+
+  def navigate_to_search_from_landing_page
+    @browser.a(:text => 'Search').when_present.click
+    Kernel.puts "Let's search some classes"
+  end
+
+  def iframe
+    # Encapsulate browser with iframe
+    @browser.iframe(:id => "ptifrmtgtframe").wait_until_present
```

Comment:
nit: Let's try and stay consistent with hashmaps and use `id: "..."` instead of the old hashrocket method.